server/src/models/userModel.js

```diff
@@ -10,7 +10,22 @@ const userSchema = new Schema(
     },
     email: {
       type: String,
-      required: true,
+      lowercase: true,
+      match: [/(.+)@(.+){2,}\.(.+){2,}/, "Please enter a valid email"],
+      validate: {
+        validator: async function (email) {
+          const user = await this.constructor.findOne({ email });
```

Comment:
We need to account for captials and lowercase. Emails are not case sensitive. 
`test@example.com` and `Test@example.com` would be treated differently here but are the same email address

Reply:
Solution I can think of is we always do `.tolowercase()` on any user input. So before saving to db we lowercase it and before comparing for login or whether or not the email already exists we lowercase it again

Reply:
@mannan0819  I believe `lowercase: true`, which is already added, should be able to take care of it.

Reply:
Sweet! Mongoose validation is sick!